### PR TITLE
Issue 5350 android archive mount fix

### LINKED ISSUE #5350: The sys.reboot() fails on Android
There seems to be a crash in `mount_android.cpp:49` after a reboot, when mapping the `game.arcd`.
The crash seems to be inside the `AAsset_getBuffer()`call:

>out_map = (void*)AAsset_getBuffer((AAsset*)out_asset);

### PR DESCRIPTION
**Pull request by JCash**

### CODE REVIEW

File: engine/liveupdate/src/liveupdate.cpp [hunk 1]
```diff
@@ -239,6 +239,11 @@ namespace dmLiveUpdate
     Result VerifyManifestReferences(const dmResource::Manifest* manifest)
     {
         assert(g_LiveUpdate.m_ResourceFactory != 0);
+        const dmResource::Manifest* base_manifest = dmResource::GetManifest(g_LiveUpdate.m_ResourceFactory);
+        // The reason the base manifest is zero here, could be that we're running from the editor.
+        // (The editor doesn't bundle the resources required for this step)
+        if (!base_manifest)
```
Comment (JCash) [inline engine/liveupdate/src/liveupdate.cpp:245]: It previously crashed when loading a manifest from an instance running from the editor.

File: engine/resource/src/mount/mount_android.cpp [hunk 2]
```diff
@@ -132,6 +130,7 @@ namespace dmResource
         }
 
         // Map index asset (if bundled) or file (if in local storage)
+        bool mem_mapped_index = false;
```
Comment (JCash) [inline engine/resource/src/mount/mount_android.cpp:133]: Let's keep track of how the resource was retrieved.

File: engine/resource/src/mount/mount_android.cpp [hunk 3]
```diff
@@ -151,61 +150,24 @@ namespace dmResource
                 dmLogError("Error mapping liveupdate index file, result = %i", r);
                 return RESULT_IO_ERROR;
             }
-        }
```
Comment (JCash) [inline engine/resource/src/mount/mount_android.cpp:154]: Removed old unused code

File: engine/resource/src/mount/mount_generic.cpp [hunk 4]
```diff
@@ -54,8 +54,14 @@ namespace dmResource
         return RESULT_OK;
     }
 
-    void UnmountArchiveInternal(dmResourceArchive::HArchiveIndexContainer &archive, void* mount_info)
+    void UnmountArchiveInternal(dmResourceArchive::HArchiveIndexContainer& archive, void* mount_info)
     {
+        dmResourceArchive::ArchiveFileIndex* afi = archive->m_ArchiveFileIndex;
+        if (afi && afi->m_FileResourceData)
```
Comment (JCash) [inline engine/resource/src/mount/mount_generic.cpp:60]: Match the create/destroy functionality in the mount/unmount functions

File: engine/resource/src/resource_archive.cpp [hunk 5]
```diff
@@ -593,12 +581,6 @@ namespace dmResourceArchive
             {
                 fclose(afi->m_FileResourceData);
             }
-
```
Comment (JCash) [inline engine/resource/src/resource_archive.cpp:596]: The actual bug: We didn't let the Unmount-function handle this, and instead incorrectly assumed it should be released using "UnmapFile".

**Review by britzl — APPROVED** [review]

**Review by britzl — APPROVED** [review]

**Pull request merged**